### PR TITLE
RFID reader

### PR DESCRIPTION
**Pull request by andrewn**
Adds support for an MFC522 RFID reader connected to the Pi's SPI interface. See `services/rfid/README.md` for wiring information.

This spike was about connected an RFID reader into the system directly on the Pi. This is probably the simplest way, the service published `rfidPresented`, `rfidRemoved` topics with the RFID's UID in the payload and an `rfidError` topic with an error message.

### CODE REVIEW

File: services/rfid/lib/rfid.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+let mfrc522 = null;
+
+const DEBUG = process.env.DEBUG;
+
+// Try to handle not running on a Pi more gracefully
```
Comment (pixelblend) [inline services/rfid/lib/rfid.js]: Is there a use in running this when the RFID library can't be found? The stubbing seems a bit redundant.

File: services/rfid/lib/rfid.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  mfrc522 = {
+    initWiringPi: () => {},
+    reset: () => {},
+    findCard: () => ({ status: '💯' }),
+    getUid: () => ({ status: '💯', data: new Buffer('FAKE') })
+  };
+}
+
+// Init WiringPi with SPI Channel 0
+mfrc522.initWiringPi(0);
```
Comment (pixelblend) [inline services/rfid/lib/rfid.js]: I think that anything that isn't purely setup should be executed in the exported function.

File: services/rfid/lib/rfid.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+// This loop keeps checking for chips. If one is near it will get the UID and authenticate
+console.log('Scanning...');
```
Comment (pixelblend) [inline services/rfid/lib/rfid.js]: It's not scanning yet... 🤔 

Could this be moved into the `scan` function?

File: services/rfid/lib/rfid.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+console.log('Please put chip or keycard in the antenna inductive zone!');
+console.log('Press Ctrl-C to stop.');
+
+const CARD_QUERY_INTERVAL_MS = 500;
+
+/*
+
```
Comment (pixelblend) [inline services/rfid/lib/rfid.js]: Was there supposed to be a comment here? 😉

File: services/rfid/lib/rfid.js (hunk 1 above)
Comment (andrewn) [reply services/rfid/lib/rfid.js]: I've mocked the library to emit a UID for testing. But yeah, maybe not much point without an interactive way to customise the fake UIDs being published.

Comment (pixelblend) [reply services/rfid/lib/rfid.js]: `DEBUG` could send out a message of a test read over websockets?

**Pull request merged**